cmd/snap-install/volmgr/sfdisk_test.go

```diff
@@ -0,0 +1,194 @@
+	c.Assert(pv.Structure[1].Name, Equals, "Recovery")
+	c.Assert(pv.Structure[1].Size, Equals, gadget.Size(0x4b000000))
+	c.Assert(pv.Structure[1].Label, Equals, "ubuntu-seed")
+	c.Assert(pv.Structure[1].Type, Equals, "C12A7328-F81F-11D2-BA4B-00A0C93EC93B")
+	c.Assert(pv.Structure[1].Filesystem, Equals, "vfat")
+}
+
+func (s *volmgrTestSuite) TestDeviceInfoNotSectors(c *C) {
+	cmdSfdisk := testutil.MockCommand(c, "sfdisk", `echo '{
+   "partitiontable": {
+      "label": "gpt",
+      "id": "9151F25B-CDF0-48F1-9EDE-68CBD616E2CA",
+      "device": "/dev/node",
+      "unit": "not_sectors",
+      "firstlba": 34,
+      "lastlba": 8388574,
+      "partitions": [
+         {"node": "/dev/node1", "start": 2048, "size": 2048, "type": "21686148-6449-6E6F-744E-656564454649", "uuid": "2E59D969-52AB-430B-88AC-F83873519F6F", "name": "BIOS Boot"}
+      ]
+   }
+}'`)
+	defer cmdSfdisk.Restore()
+
+	sf := volmgr.NewSFDisk("/dev/node")
+	_, err := sf.DeviceInfo()
+	c.Assert(err, ErrorMatches, "cannot position partitions: unknown unit .*")
+}
+
+func (s *volmgrTestSuite) TestDeviceInfoFilesystemInfoError(c *C) {
+	cmdSfdisk := testutil.MockCommand(c, "sfdisk", `echo '{
+   "partitiontable": {
+      "label": "gpt",
+      "id": "9151F25B-CDF0-48F1-9EDE-68CBD616E2CA",
+      "device": "/dev/node",
+      "unit": "sectors",
+      "firstlba": 34,
+      "lastlba": 8388574,
+      "partitions": [
+         {"node": "/dev/node1", "start": 2048, "size": 2048, "type": "21686148-6449-6E6F-744E-656564454649", "uuid": "2E59D969-52AB-430B-88AC-F83873519F6F", "name": "BIOS Boot"}
+      ]
+   }
+}'`)
+	defer cmdSfdisk.Restore()
+
+	cmdLsblk := testutil.MockCommand(c, "lsblk", "echo lsblk error; exit 1")
+	defer cmdLsblk.Restore()
+
+	sf := volmgr.NewSFDisk("/dev/node")
+	_, err := sf.DeviceInfo()
+	c.Assert(err, ErrorMatches, "cannot obtain filesystem information: lsblk error")
+}
+
+func (s *volmgrTestSuite) TestDeviceInfoJsonError(c *C) {
+	cmd := testutil.MockCommand(c, "sfdisk", `echo 'This is not a json'`)
+	defer cmd.Restore()
+
+	sf := volmgr.NewSFDisk("/dev/node")
+	info, err := sf.DeviceInfo()
+	c.Assert(err, ErrorMatches, "cannot parse sfdisk output: invalid .*")
+	c.Assert(info, IsNil)
+}
+
+func (s *volmgrTestSuite) TestDeviceInfoError(c *C) {
+	cmd := testutil.MockCommand(c, "sfdisk", "echo 'sfdisk: not found'; exit 127")
+	defer cmd.Restore()
+
+	sf := volmgr.NewSFDisk("/dev/node")
+	info, err := sf.DeviceInfo()
+	c.Assert(err, ErrorMatches, "sfdisk: not found")
+	c.Assert(info, IsNil)
+}
+
+func (s *volmgrTestSuite) TestFilesystemInfo(c *C) {
+	cmd := testutil.MockCommand(c, "lsblk", `echo '{
+   "blockdevices": [
+      {"name": "loop8p2", "fstype": "vfat", "label": "ubuntu-seed", "uuid": "C1F4-CE43", "mountpoint": null}
+   ]
+}'`)
+	defer cmd.Restore()
+
+	info, err := volmgr.FilesystemInfo("/dev/node")
+	c.Assert(cmd.Calls(), DeepEquals, [][]string{
+		{"lsblk", "--fs", "--json", "/dev/node"},
+	})
+	c.Assert(err, IsNil)
+	bd := info.BlockDevices[0]
```

Comment:
We probably want a check for the expected length of info.BlockDevices here too.

Reply:
Done.